code/controllers/subsystem/aura.dm

```diff
@@ -124,5 +155,7 @@ SUBSYSTEM_DEF(aura)
 			continue
 		if(potential_hearer.faction != faction)
 			continue
+		if(!(potential_hearer.xeno_caste.caste_flags & CASTE_FIRE_IMMUNE) && potential_hearer.on_fire) //Xenos on fire cannot receive pheros.
```

Comment:
Won't a fire immune caste never be on fire?

Reply:
I took the check handling from handle_fire's phero suppression. I suppose it could happen if something modified caste flags while on fire, but that usually shouldn't happen, yeah